Allow access to control the rectangle features needed before automatic capturing

msztech opened the pull request:
- Exposed the property to allow the control on how many features are needed before the scanner automatically take a capture. (Addressing suggestion #16)
- Updated the README documentation to reflect this change.

msztech committed:
Added Sharing Button in the example

Like described in issue #14, added a sharing button in the example using UIActivityViewController

fbernutz committed:
Update Example/DocumentScannerApp/ViewController.swift

Omitted the unnecessary parameter

Co-Authored-By: msztech <23656610+msztech@users.noreply.github.com>

msztech committed:
Delete .DS_Store

msztech committed:
Exposed the featuresRequired variable and added its explanation in the README file.

msztech committed:
Merge remote-tracking branch 'upstream/master'

Merge remote changes to local

fbernutz reviewed (changes requested):
Hi @msztech 👋 
Could you please resolve the merge conflict with updating your fork with the latest master (like [this](https://gist.github.com/CristinaSolana/1885435)). 

As proposed in #16 we would like to choose an enum to configure the quality with the number of required features. I think it would be a difficult API when a user has to choose a number of required features as Integer. We thought about configuring this with an enum, which has three cases (high, medium and fast). 

So instead of `var featuresRequired: Int { get set }` we thought about having `var quality: Quality { get set }`. 
This enum can look like this: 

```swift
enum Quality {
  case high, medium, fast 
}
```

Then you have to define some values for which concrete number of required features the quality of a scanned image is high, medium or low. 

With setting the quality for the scanner, I think, it is much easier to assess which result can be expected. 

I hope you understand, what @xavierLowmiller and me would like you to change. If you have any questions, feel free to ask! We really appreciate your work!

msztech commented:
@fbernutz Thanks for the suggestions. I’ll resolve the conflicts and modify the code to make it easier to control the scanning quality. By the way, is there specific features required for different levels of quality? Should I experimenting with it and come up with some data?

fbernutz commented:
There is nothing specified yet. Would be nice, if you could experiment with it and make a suggestion for the different levels 😊Thanks!

msztech commented:
@fbernutz Sounds great! Thanks! 👍

msztech committed:
Merge branch 'master' of https://github.com/adorsys/document-scanner-ios

msztech committed:
scanningQuality control variable

- Added an enum `featuresRequired` that allows the control of the number of features required before the automatic capture. Experimented with the value. (When it's .fast, it takes a picture once it's stabilized.)
- Fixed the .pbxproj since there're some file name changes and it's not reflected in the project.
- Fixed the merge conflict in the README, updated the file to contain the new variable.

msztech commented:
@fbernutz @xavierLowmiller I have resolved the conflict and revised the feature. Please help me review the changes. By the way, nice new icon for the example app! 👍

fbernutz commented:
Looks good to me and works as expected! 

The only thing I am thinking about is the naming of the enum cases. It now contains `high, medium and fast`. When changing the enum in the example app, I am confused whether `fast` or `high` results in a better quality, because both names have a positive connotation. I know, we suggested these cases but I am not sure about them anymore. Would be `high, medium and low` better cases? 
@xavierLowmiller what do you think?

xavierLowmiller commented:
The idea for `fast` was that `low` sounds quite negative ("Obviously I'm choosing high quality, what idiot would set this to `low`"?), So I'm good with the naming.

I think we can merge this and get more feedback later

fbernutz commented:
So, this is merged now. Will make a release later. 🎉 
Thanks again, @msztech, for your invested time and this nice feature ❤️

fbernutz commented:
A new version 1.2.0 with this change is now available in CocoaPods.
